src/pip/_internal/operations/install/wheel.py

```diff
@@ -563,28 +564,25 @@ def is_entrypoint_wrapper(name):
             logger.warning(msg)
 
     # Record pip as the installer
-    installer = os.path.join(dest_info_dir, 'INSTALLER')
-    temp_installer = os.path.join(dest_info_dir, 'INSTALLER.pip')
-    with open(temp_installer, 'wb') as installer_file:
+    installer_path = os.path.join(dest_info_dir, 'INSTALLER')
+    with adjacent_tmp_file(installer_path) as installer_file:
         installer_file.write(b'pip\n')
-    shutil.move(temp_installer, installer)
-    generated.append(installer)
+    replace(installer_file.name, installer_path)
+    generated.append(installer_path)
 
     # Record details of all files installed
-    record = os.path.join(dest_info_dir, 'RECORD')
-    temp_record = os.path.join(dest_info_dir, 'RECORD.pip')
-    with open_for_csv(record, 'r') as record_in:
-        with open_for_csv(temp_record, 'w+') as record_out:
-            reader = csv.reader(record_in)
-            outrows = get_csv_rows_for_installed(
-                reader, installed=installed, changed=changed,
-                generated=generated, lib_dir=lib_dir,
-            )
-            writer = csv.writer(record_out)
-            # Sort to simplify testing.
-            for row in sorted_outrows(outrows):
-                writer.writerow(row)
-    shutil.move(temp_record, record)
+    record_path = os.path.join(dest_info_dir, 'RECORD')
+    with open(record_path, **csv_io_kwargs('r')) as record_file:
+        rows = get_csv_rows_for_installed(
+            csv.reader(record_file),
+            installed=installed,
+            changed=changed,
+            generated=generated,
+            lib_dir=lib_dir)
+    with adjacent_tmp_file(record_path, **csv_io_kwargs('w+')) as record_file:
```

Comment:
Doesn't it work with plain `'w'` mode ?

Reply:
I don't know, I asked the same question when I open this PR.  From the code then no,  I don't think CSV needs either seek or read at all.

Reply:
It might be a left-over from a time where the RECORD file was opened to append a line at the end ?

Reply:
I'd imagine yes, but `w+` was in the original code, so I'd be OK with fixing this in a follow-up if @McSinyx prefers.

Reply:
> I'd be OK with fixing this in a follow-up if @McSinyx prefers.

Sorry I'm not sure if I get what you mean.  Is it that it's OK to switch too the less confusing `w` mode *now* or a separate follow-up PR?

Reply:
Sorry - as I say I was responding without having seen the full conversation.

If you're happy to do so, I suggest we switch to `w` now.

Reply:
My pleasure! Convenient to work around the stuck Travis too!